### ISSUE TITLE
yamldecode does not reserve case for "True" and "False" strings

### ISSUE THREAD

**Issue by houqp** [hashicorp/terraform#22138]
### Terraform Version

```
Terraform v0.12.5
+ provider.aws v2.16.0
+ provider.local v1.2.2
+ provider.null v2.1.2
+ provider.template v2.1.2
```

### Steps to Reproduce

This bug can be easily reproduced within console:

``` bash
 $ terraform console
> yamldecode("- 'True'")
[
  "true",
]
```

It should return the following instead:

```bash
> yamldecode("- 'True'")
[
  "True",
]
```

**Comment by houqp** [comment]
Can confirm this is an upstream bug: https://github.com/zclconf/go-cty-yaml/issues/3

**Comment by apparentlymart** [comment]
The upstream fix is now merged, in #22258. Thanks for finding and fixing this, @houqp!
